### ISSUE TITLE
Issues with "weird" deps

### ISSUE THREAD

**Issue by antipax** [bitwalker/exrm#249]
Hi... it's me again... :p

For the JSX library:
```
** (UndefinedFunctionError) undefined function: JSX.Mixfile.application/0
    JSX.Mixfile.application()
    lib/exrm/deps.ex:176: anonymous fn/2 in ReleaseManager.Deps.get_applications/1
    (mix) lib/mix/project.ex:196: Mix.Project.in_project/4
    (elixir) lib/file.ex:1137: File.cd!/2
    (elixir) lib/enum.ex:1043: anonymous fn/3 in Enum.map/2
    (elixir) lib/enum.ex:1387: Enum."-reduce/3-lists^foldl/2-0-"/3
    (elixir) lib/enum.ex:1043: Enum.map/2
    lib/exrm/deps.ex:140: ReleaseManager.Deps.get_included_applications/1
```
this seems to be because the Mixfile is actually defined as:
```elixir
defmodule EXJSX.Mixfile do
  ...
end
```

and for the Mock library (which we are looking to remove because mocks are bad, mmkay, but it's still here for now):
```
** (UndefinedFunctionError) undefined function: Mock.Mixfile.application/0
    Mock.Mixfile.application()
    lib/exrm/deps.ex:176: anonymous fn/2 in ReleaseManager.Deps.get_applications/1
    (mix) lib/mix/project.ex:196: Mix.Project.in_project/4
    (elixir) lib/file.ex:1137: File.cd!/2
    (elixir) lib/enum.ex:1043: anonymous fn/3 in Enum.map/2
    (elixir) lib/enum.ex:1387: Enum."-reduce/3-lists^foldl/2-0-"/3
    (elixir) lib/enum.ex:1043: Enum.map/2
    lib/exrm/deps.ex:140: ReleaseManager.Deps.get_included_applications/1
```
which seems to be because the Mock mixfile does not include a definition for `application/0`.

These are both during mix release on latest master.

**Comment by bitwalker** [comment]
@antipax Hey there! So yeah I literally just pushed a fix for the second issue, but the first one is a little surprising. The module for the mixfile is being fetched by Mix.Project.get!, so it should be correct. Could you try updating to the latest exrm (1.0.0-rc4 now, or master if you are using git directly) and try again?

**Comment by antipax** [comment]
@bitwalker yup, still seeing the same error. It seems there's a "jsx" mix package which is for the erlang package, which uses the JSX.Mixfile module name, and an "exjsx" mix package which is an elixir wrapper for the erlang package, which uses EXJSX.Mixfile for its module name.

**Comment by bitwalker** [comment]
@antipax Are you still seeing the error for Mock as well? Since `jsx` has a mixfile, and it doesn't define `application/0` which was the same issue which caused Mock to fail, the fix I pushed should have resolved it for both. If you are still seeing it just for `jsx`, I'll have to dig a little deeper to figure out what's causing the problem.

**Comment by antipax** [comment]
No error for mock. Looks like the package causing issues now is "exjsx", I think jsx itself is fine.

**Comment by antipax** [comment]
I will check tomorrow morning though, just to confirm.

**Comment by bitwalker** [comment]
I'll set up a test project and see if I can reproduce, probably won't be until later tonight/tomorrow morning though.

**Comment by antipax** [comment]
https://gist.github.com/antipax/1891c0419d71ae76679d

**Comment by antipax** [comment]
Oh, also it seems my project is failing for Mock still. Not sure why that could be happening.

**Comment by jeffweiss** [comment]
I've run into the same error, but with a different dependency (which also is just a library, not an application): https://github.com/meh/jazz.
I've created a test repo (https://github.com/jeffweiss/exrm_test) which successfully creates a release using exrm 0.19.9 and conform 0.17.0 (`master` branch), but fails for the same reason as @antipax sees w/ `jsx` when using exrm 1.0.0-rc4 and conform 1.0.0-rc2 (`not_working` branch).

when running `MIX_ENV=prod mix release` on the `not_working` branch this is error I see: 
```
Building release with MIX_ENV=prod.
** (UndefinedFunctionError) undefined function: Jazz.Mixfile.application/0
    Jazz.Mixfile.application()
    lib/exrm/deps.ex:183: anonymous fn/2 in ReleaseManager.Deps.get_applications/1
    (mix) lib/mix/project.ex:196: Mix.Project.in_project/4
    (elixir) lib/file.ex:1137: File.cd!/2
    (elixir) lib/enum.ex:1043: anonymous fn/3 in Enum.map/2
    (elixir) lib/enum.ex:1387: Enum."-reduce/3-lists^foldl/2-0-"/3
    (elixir) lib/enum.ex:1043: Enum.map/2
    lib/exrm/deps.ex:147: ReleaseManager.Deps.get_included_applications/1
```

**Comment by bitwalker** [comment]
Found the problem, I was an idiot and had duplicated some logic that was bad. New release with the fix is 1.0.0-rc5
